autoemulate/experimental/data/utils.py

```diff
@@ -190,6 +192,28 @@ def _denormalize(
         return (x * x_std) + x_mean
 
 
+class RandomMixin:
+    """Mixin class for ensuring reproducibility in AutoEmulate models."""
+
+    def set_random_seed(self, seed: int = 42, deterministic: bool = True):
```

Comment:
Perhaps it makes sense for this to remain a `@staticmethod` since `self` is not referred to in the method?